qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -158,50 +151,45 @@ def cancel(self):
-        job_result = self._api.get_job(self._job_id)
-        stats = {'job_id': self._job_id}
-        self._status = None
-        _status_msg = None
-        if 'status' not in job_result:
-            self._exception = QISKitError("get_job didn't return status: %s" %
-                                          (pprint.pformat(job_result)))
-            raise QISKitError("get_job didn't return status: %s" %
-                              (pprint.pformat(job_result)))
-        elif job_result['status'] == 'RUNNING':
+        """ Query the API for the status of the Job """
+        self._wait_for_submitting()
+        api_job = self._api.get_job(self._job_id)
+        if 'status' not in api_job:
+            raise IBMQJobError("get_job didn't return status: %s" %
+                               (pprint.pformat(api_job)))
+        elif api_job['status'] == 'RUNNING':
             self._status = JobStatus.RUNNING
             # we may have some other information here
-            if 'infoQueue' in job_result:
-                if 'status' in job_result['infoQueue']:
-                    if job_result['infoQueue']['status'] == 'PENDING_IN_QUEUE':
+            if 'infoQueue' in api_job:
+                if 'status' in api_job['infoQueue']:
+                    if api_job['infoQueue']['status'] == 'PENDING_IN_QUEUE':
                         self._status = JobStatus.QUEUED
-                if 'position' in job_result['infoQueue']:
-                    stats['queue_position'] = job_result['infoQueue']['position']
-        elif job_result['status'] == 'COMPLETED':
+                if 'position' in api_job['infoQueue']:
+                    self._queue_position = api_job['infoQueue']['position']
+        elif api_job['status'] == 'COMPLETED':
             self._status = JobStatus.DONE
-        elif job_result['status'] == 'CANCELLED':
+        elif api_job['status'] == 'CANCELLED':
+            self._cancelled = True
             self._status = JobStatus.CANCELLED
-        elif self.exception or self._future_submit.exception():
-            self._status = JobStatus.ERROR
-            if self._future_submit.exception():
-                self._exception = self._future_submit.exception()
-            self._status_msg = str(self.exception)
-        elif 'ERROR' in job_result['status']:
+        elif 'ERROR' in api_job['status']:
             # ERROR_CREATING_JOB or ERROR_RUNNING_JOB
             self._status = JobStatus.ERROR
-            self._status_msg = job_result['status']
+            self._status_msg = api_job['status']
         else:
             self._status = JobStatus.ERROR
             raise IBMQJobError('Unexpected behavior of {0}\n{1}'.format(
                 self.__class__.__name__,
-                pprint.pformat(job_result)))
-        stats['status'] = self._status
-        stats['status_msg'] = _status_msg
-        return stats
+                pprint.pformat(api_job)))
+        return self._status
+
+    def queue_position(self):
```

Comment:
this is a useful method to have. I also think it is useful to have a public method for getting `_creation_date`

Reply:
I think this is available as job.creation_date, a job attribute string.